death-star-client/src/main/kotlin/com/example/deathstarclient/DeathStarClient.kt

```diff
@@ -22,70 +19,59 @@ class DeathStarClient {
         val planetsStream = deathStarStub.destroy()
-        val logStream = logStub.newUser(LogServiceProto.User.newBuilder()
-                .setName(userName)
-                .build())
-        val scoresStream = scoreStub.scores(Empty.getDefaultInstance())
+        val logStream = logStub.newUser{ name = userName }
+        val scoresStream = scoreStub.scores()
         println("Received all streams")
-        return JoinResult(/*krotoPlanetsStream, */planetsStream, logStream, scoresStream)
+
+        return JoinResult(planetsStream, logStream, scoresStream)
     }
 
     data class JoinResult (
-            //val krotoPlanetsStream: ClientBidiCallChannel<PlanetProto.DestroyPlanetRequest, PlanetProto.Planets>,
-            val planetsStream: ManyToManyCall<PlanetProto.DestroyPlanetRequest, PlanetProto.Planets>,
-            val logStream: ReceiveChannel<LogServiceProto.Log>,
-            val scoresStream: ReceiveChannel<ScoreServiceProto.ScoresResponse>
+        val planetsStream: ClientBidiCallChannel<PlanetProto.DestroyPlanetRequest, PlanetProto.Planets>,
+        val logStream: ReceiveChannel<LogServiceProto.Log>,
+        val scoresStream: ReceiveChannel<ScoreServiceProto.ScoresResponse>
     )
 
     fun succesfulDestroyAttempt(p: PlanetProto.Planet): Boolean {
         val prob = 0.1 * Math.sqrt(260412.5 - (p.weight * p.weight))
         return Random.nextInt(0..100) < prob
     }
 
-    suspend fun tryDestroy(planet: PlanetProto.Planet, name: String, inChannel: SendChannel<PlanetProto.DestroyPlanetRequest>, x: Int, y: Int) {
+    suspend fun tryDestroy(
+        planet: PlanetProto.Planet,
+        name: String,
+        inChannel: SendChannel<PlanetProto.DestroyPlanetRequest>,
+        x: Int, y: Int
+    ) {
         if (succesfulDestroyAttempt(planet)) {
-            inChannel.send(DestroyPlanetRequest {
+            inChannel.send {
                 userName = name
                 planetId = planet.planetId
                 weight = planet.weight
-                coordinates = Coordinates {
+                coordinates {
                     this.x = x
                     this.y = y
-
                 }
-            })
+            }
         }
     }
 
-    fun DestroyPlanetRequest(init: PlanetProto.DestroyPlanetRequest.Builder.() -> Unit) =
-            PlanetProto.DestroyPlanetRequest.newBuilder()
-                    .apply(init)
-                    .build()
-
-    fun Coordinates(init: PlanetProto.Coordinates.Builder.() -> Unit) =
-            PlanetProto.Coordinates.newBuilder()
-                    .apply(init)
-                    .build()
-
-
     private fun channelForTarget(target: String): ManagedChannel {
         return ManagedChannelBuilder
                 .forTarget(target)
                 .nameResolverFactory(DnsNameResolverProvider())
-                .loadBalancerFactory(RoundRobinLoadBalancerFactory.getInstance())
+                .defaultLoadBalancingPolicy("round_robin")
```

Comment:
This was only done because the previous method was removed in the latest version of grpc